File: 算法与数据结构/golang实现数据结构/DataStructAndAlgorithm/set/bstset/bstset.go
package bstset

import "DataStructAndAlgorithm/bst/bst"

type BstSet struct {
	bst *bst.Bst
}

// 新建BstSet对象实例
func NewBstSet() *BstSet {
	return &BstSet{bst: bst.NewBst()}
}
// 添加集合元素
func (s BstSet) Add(e interface{}) {
	s.bst.Add(e)
}

// 移除集合元素
func (s BstSet) Remove(e interface{}){
	s.bst.Remove(e)
}

// 判断集合是否包含e元素
func (s BstSet) Contains(e interface{}) bool {
	return s.bst.Contains(e)
}

// 获取集合元素个数
func (s BstSet) GetSize() int {
	return s.bst.GetSize()
}

// 判断集合是否为空
func (s BstSet)	IsEmpty() bool {
	return s.bst.IsEmpty()
}
// 基于底层bst前序遍历实现的String(),格式化输出集合中的元素
func (s BstSet) String() string{
	s.bst.PreOrder()
	return ""
}

File: 算法与数据结构/python实现数据结构/set_/bstset.py
# coding:utf-8
from bst_.bst import Bst


class BstSet(object):
    def __init__(self):
        self.bst = Bst()

    def add(self, e):
        """
        添加集合元素
        :param e:
        :return:
        """
        self.bst.add(e)

    def remove(self, e):
        """
        移除集合元素
        :param e:
        :return:
        """
        self.bst.remove(e)

    def contains(self, e):
        """
        判断集合是否包含e元素
        :param e:
        :return:
        """
        return self.bst.contains(e)

    def get_size(self):
        """
        获取集合元素个数
        :return:
        """
        return self.bst.get_size()

    def is_empty(self):
        """
        判断集合是否为空
        :return:
        """
        return self.bst.is_empty()

    def __repr__(self):
        """
        基于底层bst前序遍历实现的string(), 格式化输出集合中的元素
        :return:
        """
        self.bst.pre_order()
        return ""


if __name__ == "__main__":
    words = ["1", "2", "2", "A", "B", "D"]
    s = BstSet()
    for i in words:
        s.add(i)
    print(s.is_empty())
    print(s.get_size())
    print(s.contains("A"))
    print(s)